### PR TITLE
[embedded] Improve the diagnostic message when using an existential

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

File: include/swift/AST/DiagnosticsSIL.def [hunk 1]
```diff
@@ -352,6 +352,10 @@ ERROR(global_must_be_compile_time_const,none,
       "global variable must be a compile-time constant", ())
 ERROR(non_final_generic_class_function,none,
       "classes cannot have non-final generic fuctions in embedded Swift", ())
+ERROR(embedded_swift_existential_type,none,
+      "Existential type %0 is unavailable in embedded Swift", (Type))
```
Comment (eeckstein) [inline include/swift/AST/DiagnosticsSIL.def:356]: nitpick: I think we don't use "existential" as user facing term. There are a few other error messages which contain "existential", but I think it's better to call it "protocol-type" or "protocol as type", like it's done for other sema error messages.

**Review by eeckstein — APPROVED** [review]
lgtm.
Just a small comment about the wording.

File: include/swift/AST/DiagnosticsSIL.def (hunk 1 above)
Comment (xedin) [reply include/swift/AST/DiagnosticsSIL.def:356]: `value of protocol type %0` is what we use in Sema.

Another nit: the message shouldn't be capitalized.

Comment (kubamracek) [reply include/swift/AST/DiagnosticsSIL.def]: How about this wording?
```
 ERROR(embedded_swift_existential_type,none,
       "using values of protocol type %0 is not allowed in embedded Swift", (Type))
 ERROR(embedded_swift_existential,none,
       "using values of protocol types is not allowed in embedded Swift", ())
 ```

**Comment by kubamracek** [comment]
@swift-ci please test

File: include/swift/AST/DiagnosticsSIL.def [hunk 2]
```diff
@@ -352,6 +352,10 @@ ERROR(global_must_be_compile_time_const,none,
       "global variable must be a compile-time constant", ())
 ERROR(non_final_generic_class_function,none,
       "classes cannot have non-final generic fuctions in embedded Swift", ())
+ERROR(embedded_swift_existential_type,none,
+      "using values of protocol type %0 is not allowed in embedded Swift", (Type))
+ERROR(embedded_swift_existential,none,
+      "using values of protocol types is not allowed in embedded Swift", ())
```
Comment (xedin) [inline include/swift/AST/DiagnosticsSIL.def:358]: How about `cannot use a value of protocol type %0 in embedded Swift`? this aligns better with the rest of the diagnostic we have.

Comment (kubamracek) [reply include/swift/AST/DiagnosticsSIL.def]: That sounds great, let's go with that. Thanks!

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test
